src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.cs

```diff
@@ -769,7 +778,7 @@ public unsafe bool IsSuffix(ReadOnlySpan<char> source, ReadOnlySpan<char> suffix
             else
             {
                 // Linguistic comparison requested and we don't need to special-case any args.
-#if TARGET_BROWSER
+#if TARGET_BROWSER || TARGET_OSX || TARGET_MACCATALYST || TARGET_IOS || TARGET_TVOS
```

Comment:
The code to set `matchLengthPtr` in `CompareInfo.ICU.cs` and `CompareInfo.OSX.cs` suggests that OSX does support match length.

Reply:
If it's supported for `IndexOf` and `LastIndexOf` then why not to use e.g. `IndexOf` inside the `IsPrefix` impelementation and if the result == 0 then return true for being the prefix + use the `matchLengthPtr` value, otherwise: false.
It OSX's implementation of `IndexOf` will surely take more computing than `IsPrefix` so we could do it only when the user actually passed a non-null pointer to `matchLengthPtr`.

Reply:
On `OSX` `IndexOf` and `LastIndexOf` linguistic comparison is not fully covered. I tried to cover all possible cases. But in cases when `source` and `prefix` will contain letters with different char lengths they will not work correctly (more about covered / not covered cases is written in doc) 
I prefer to go with current implementation and not support  `matchLengthPtr` for `IsPrefix` and `IsSuffix` as using `IndexOf` will not give correct `matchLengthPtr` for all cases and can even return`false` for some cases when it should be `true`.

In the future when all these missing cases will be covered that is a good idea to call `IndexOf` inside `IsPrefix`.

Reply:
I missed the fact that `IsPrefix` has wider linguistic support than `IndexOf`. If it's like that, then I would vote for unifying both to `IsPrefix` and removing the support for `matchLengthPtr` for IOS. When disabling `matchLengthPtr` for WASM we did a research on frequency of usage of this parameter on GitHub and it is used very rarely.
|     **API**     |                                                                         **All usages**                                                                        |                                                                                                               **Usages with matchLength**                                                                                                               |
|:---------------:|:-------------------------------------------------------------------------------------------------------------------------------------------------------------:|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------:|
|   **IndexOf**   |   [610 000](https://github.com/search?q=language%3AC%23++%2F%5C.IndexOf%2F++AND+NOT+%28path%3ACompareInfo.cs+OR+path%3AString.Manipulation.cs%29&type=code)   |   [25](https://github.com/search?q=language%3AC%23++%2F%5C.IndexOf%5C%28%5B%5E%22%28%29%5D*%2C%5B%5E%22%28%29%5D*%2C%5B%5E%22%28%29%5D*%2C%5B%5E%22%28%29%5D*+out+%2F+AND+NOT+%28path%3ACompareInfo.cs+OR+path%3AString.Manipulation.cs%29&type=code)   |
| **LastIndexOf** | [171 000](https://github.com/search?q=language%3AC%23++%2F%5C.LastIndexOf%2F++AND+NOT+%28path%3ACompareInfo.cs+OR+path%3AString.Manipulation.cs%29&type=code) | [21](https://github.com/search?q=language%3AC%23++%2F%5C.LastIndexOf%5C%28%5B%5E%22%28%29%5D*%2C%5B%5E%22%28%29%5D*%2C%5B%5E%22%28%29%5D*%2C%5B%5E%22%28%29%5D*+out+%2F+AND+NOT+%28path%3ACompareInfo.cs+OR+path%3AString.Manipulation.cs%29&type=code) |
|   **IsPrefix**  |    [2 400](https://github.com/search?q=language%3AC%23++%2F%5C.IsPrefix%2F++AND+NOT+%28path%3ACompareInfo.cs+OR+path%3AString.Manipulation.cs%29&type=code)   |   [14](https://github.com/search?q=language%3AC%23++%2F%5C.IsPrefix%5C%28%5B%5E%22%28%29%5D*%2C%5B%5E%22%28%29%5D*%2C%5B%5E%22%28%29%5D*%2C%5B%5E%22%28%29%5D*+out+%2F+AND+NOT+%28path%3ACompareInfo.cs+OR+path%3AString.Manipulation.cs%29&type=code)  |
|   **IsSuffix**  |     [804](https://github.com/search?q=language%3AC%23++%2F%5C.IsSuffix%2F++AND+NOT+%28path%3ACompareInfo.cs+OR+path%3AString.Manipulation.cs%29&type=code)    |   [12](https://github.com/search?q=language%3AC%23++%2F%5C.IsSuffix%5C%28%5B%5E%22%28%29%5D*%2C%5B%5E%22%28%29%5D*%2C%5B%5E%22%28%29%5D*%2C%5B%5E%22%28%29%5D*+out+%2F+AND+NOT+%28path%3ACompareInfo.cs+OR+path%3AString.Manipulation.cs%29&type=code)  |

I believe that from this reason, it is more important to support linguistic comparison on the highest level we can, even at the cost of blocking this API for IOS's Hybrid. We can do it in the next PR where we will measure also which approach is less computing-costly.

Reply:
I have opened https://github.com/dotnet/runtime/issues/87767 for future investigation.

Reply:
I have opened https://github.com/dotnet/runtime/issues/87767 for future investigation.